### PR TITLE
fix(attachments): fix attachment deletion for edited submissions TASK-1841

### PR DESCRIPTION
**Pull request by rajpatel24**
### 🗒️ Checklist

1. [x] run linter locally
2. [x] update developer docs (API, README, inline, etc.), if any
3. [x] for user-facing doc changes create a Zulip thread at #Kobo support docs, if any
4. [x] draft PR with a title `<type>(<scope>)<!>: <title> DEV-1234`
5. [x] assign yourself, tag PR: at least `Front end` and/or `Back end` or `workflow`
6. [x] fill in the template below and delete template comments
7. [x] review thyself: read the diff and repro the preview as written
8. [ ] open PR & confirm that CI passes & request reviewers, if needed
9. [ ] delete this section before merging

### 📣 Summary
Fixes an issue where attachments could not be deleted from edited submissions due to incorrect handling of submission UUIDs.

### 📖 Description
This fix updates the attachment deletion logic to correctly filter using `root_uuid` for edited submissions. Previously, the API returned a 400 error when edited submissions were included, treating their root UUIDs as invalid. The updated query ensures attachments from both original and edited submissions can be deleted successfully.

### CODE REVIEW

File: kpi/serializers/v2/attachment_delete.py [hunk 1]
```diff
@@ -54,9 +55,12 @@ def save(self, request, asset):
...
             field = 'attachment_uid'
         else:
+            uuids = self.validated_data['submission_root_uuids']
             attachments = Attachment.objects.filter(
-                xform_id=deployment.xform_id,
-                instance__uuid__in=self.validated_data['submission_root_uuids'],
+                Q(xform_id=deployment.xform_id) & (
+                    Q(instance__root_uuid__in=uuids) |
+                    (Q(instance__uuid__in=uuids) & Q(instance__root_uuid__isnull=True))
+                )
```
Comment (noliveleger) [inline kpi/serializers/v2/attachment_delete.py]: This could be simplified with  : 

```python 
Q(instance__root_uuid__in= uuids)
| Q(instance__uuid__in= uuids, instance__root_uuid__isnull=True),
xform_id=deployment.xform_id
```

Let's add a TODO to remove the condition when `0005` long-running migration is complete (in few months) (same for the NLP bugfix already merged please.

File: kpi/tests/api/v2/test_api_attachments_delete_viewset.py [hunk 2]
```diff
@@ -300,6 +300,53 @@ def test_bulk_delete_attachments_success(self):
...
+        """
+        Test bulk deletion of attachments for a submission that has been edited
+        """
+        self.client.login(username='someuser', password='someuser')
+        edited_submission = {
+            '__version__': self.asset.latest_deployed_version.uid,
+            'meta': {
+                'instanceID': 'uuid:test_edited_uuid',
+                'rootUuid': 'uuid:test_uuid1',
+                'deprecatedID': 'uuid:test_uuid1',
```
Comment (noliveleger) [inline kpi/tests/api/v2/test_api_attachments_delete_viewset.py:313]: We need a test for old submissions also, where `root_uuid` is null.
It can be faked with `Instance.objects.get(root_uuid=test_uuid1).update(root_uuid=None)`
You could use `ddt` to avoid rewriting another test but just use a condition to reset the root_uuid?

File: kpi/serializers/v2/attachment_delete.py (hunk 1 above)
Comment (noliveleger) [reply kpi/serializers/v2/attachment_delete.py]: @rajpatel24 , please use the simplified version for the filter ;-)

**Review by sksondra — COMMENTED** [review]
Good from QA.
